<?xml version="1.0"?>
<launch>
  <arg name="camera_topic" default="/camera/image_raw"/>
  <arg name="camera_info_topic" default="/camera/camera_info"/>
  <arg name="debug" default="true"/>

  <!-- Model path argument -->
  <arg name="model_path" default="$(find road_segmentation)/models/pspunet_weight.h5"/>

  <!-- Load the class.yaml file contents into ROS parameter server -->
  <rosparam file="$(find road_segmentation)/config/class.yaml" command="load"/>

  <node pkg="image_proc" type="image_proc" ns="/insta360/front" name="image_proc"/>
  <node pkg="nodelet" type="nodelet" args="standalone image_proc/crop_decimate" name="my_decimator">
    <param name="x_offset" type="int" value="155"/>
    <param name="y_offset" type="int" value="248"/>
    <!-- <param name="width" type="int" value="533"/> -->
    <!-- <param name="height" type="int" value="300"/> -->
    <param name="width" type="int" value="426"/>
    <param name="height" type="int" value="240"/>

    <!-- remap input topics -->
    <remap from="camera/image_raw" to="/insta360/front/image_rect_color"/>
    <remap from="camera/image_info" to="/insta360/front/camera_info"/>

    <!-- remap output topics -->
    <remap from="camera_out/image_raw" to="/insta360/front/cropped/image"/>
    <remap from="camera_out/image_info" to="/insta360/front/cropped/camera_info"/>
  </node>

  <!-- Node execution -->
  <node pkg="road_segmentation" type="road_segmentation_node.py" name="road_segmentation_node"
    output="screen">
    <param name="camera_topic" value="$(arg camera_topic)"/>
    <param name="camera_info_topic" value="$(arg camera_info_topic)"/>
    <param name="model_path" value="$(arg model_path)"/>
    <param name="debug" value="$(arg debug)"/>
  </node>
</launch>
